File: Leetcode_POTD_2025/10_OCTOBER/2598_Smallest_Missing_Non_negative_Integer_After_Operations.py
class Solution(object):
    def findSmallestInteger(self, input_data, modulus_key):
        tally_array = [0] * modulus_key
        for item in input_data:
            tally_array[item % modulus_key] += 1
        
        max_chain = 0
        while tally_array[max_chain % modulus_key] > 0:
            tally_array[max_chain % modulus_key] -= 1
            max_chain += 1
            
        return max_chain

File: Leetcode_POTD_2025/10_OCTOBER/2598_Smallest_Missing_Non_negative_Integer_After_Operations.cpp
class Solution {
public:
    int findSmallestInteger(std::vector<int>& numz, int v_base) {
        std::vector<int> residues(v_base, 0);
        for (int n : numz) {
            residues[(n % v_base + v_base) % v_base]++;
        }
        int m_val = 0;
        while (residues[m_val % v_base]-- > 0) m_val++;
        return m_val;
    }
};
